valhalla/meili/match_result.h

```diff
@@ -59,20 +59,13 @@ struct MatchResults {
     e2 = this->segments.empty() || this->segments.back().target > 0.0f ? edges.cend()
                                                                        : edges.cend() - 1;
   }
+  // equality here means that the sequence of edges within p either contains or is contained in
+  // this's sequence of edges, so we do basically a substring search algorithm
   bool operator==(const MatchResults& p) const {
```

Comment:
so yeah i dont know why when i originally implemented this that i felt the need to write a subpart substring search algorithm which was throwing an ASAN error. instead of that idiocy i am just using std::search which does the hard work for me. the docs didnt say where the implementation is robust to different sizes of ranges being searched so i put that in my logic below just in case (i know other stl methods dont protect against it so its there just in case).